interfaces/builtin/kubernetes_support.go

```diff
@@ -35,65 +47,227 @@ const kubernetesSupportBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
-const kubernetesSupportConnectedPlugAppArmor = `
-# Description: can use kubernetes to control Docker containers. This interface
-# is restricted because it gives wide ranging access to the host and other
-# processes.
+const kubernetesSupportConnectedPlugAppArmorCommon = `
+# Common rules for running as a kubernetes node
 
-# what is this for?
-#include <abstractions/dbus-strict>
-
-# Allow reading all information regarding cgroups for all processes
+# reading cgroups
 capability sys_resource,
-@{PROC}/@{pid}/cgroup r,
 /sys/fs/cgroup/{,**} r,
 
-# Allow adjusting the OOM score for Docker containers. Note, this allows
-# adjusting for all processes, not just containers.
+# Allow adjusting the OOM score for containers. Note, this allows adjusting for
+# all processes, not just containers.
 @{PROC}/@{pid}/oom_score_adj rw,
-/sys/kernel/mm/hugepages/ r,
+@{PROC}/sys/vm/overcommit_memory rw,
+/sys/kernel/mm/hugepages/{,**} r,
+
+capability dac_override,
+
+/usr/bin/systemd-run Cxr -> systemd_run,
+profile systemd_run (attach_disconnected,mediate_deleted) {
```

Comment:
Since we are not otherwise using `mediate_deleted` would it make sense to explain why it is used here? After some googling the best I could find was https://gitlab.com/apparmor/apparmor/wikis/FAQ